core/src/database/DatabaseSessionPool.cpp

```diff
@@ -28,17 +28,50 @@
  * SOFTWARE.
  *
  */
+
 #include "DatabaseSessionPool.hpp"
 #include "migrations.hpp"
 
 namespace ledger {
     namespace core {
+        DatabaseSessionPool::DatabaseSessionPool(
+            const std::shared_ptr<DatabaseBackend> &backend,
+            const std::shared_ptr<api::PathResolver> &resolver,
+            const std::shared_ptr<spdlog::logger>& logger,
+            const std::string &dbName
+        ) : _pool((size_t) backend->getConnectionPoolSize()), _buffer("SQL", logger) {
+            if (logger != nullptr && backend->isLoggingEnabled()) {
+                _logger = new std::ostream(&_buffer);
+            } else {
+                _logger = nullptr;
+            }
+
+            auto poolSize = backend->getConnectionPoolSize();
+            for (size_t i = 0; i < poolSize; i++) {
+                auto& session = getPool().at(i);
+                backend->init(resolver, dbName, session);
+                if (_logger != nullptr)
+                    session.set_log_stream(_logger);
+            }
+
+            // Migrate database
+            performDatabaseMigration();
+        }
+
+        DatabaseSessionPool::~DatabaseSessionPool() {
+            if (_logger != nullptr) {
```

Comment:
This is redundant, it is safe to `delete nullptr;`

Reply:
Ah this is not a new code... ok then

Reply:
Yeah I also felt it could be removed. I will remove it. Cleaning codebase is important.